tests/devices/test_default_qutrit_mixed.py

```diff
@@ -33,8 +93,929 @@ def test_debugger_attribute():
+        import torch
+
+        phi = torch.tensor(-0.526, requires_grad=True)
+
+        dev = DefaultQutritMixed()
+
+        def f(x):
+            qs = self.get_TRX_quantum_script(x, subspace)
+            return dev.execute(qs)
+
+        result = f(phi)
+        expected = expected_TRX_circ_expval_values(phi.detach().numpy(), subspace)
+
+        result_detached = math.asarray(result, like="torch").detach().numpy()
+        assert math.allclose(result_detached, expected)
+
+        jacobian = math.asarray(torch.autograd.functional.jacobian(f, phi + 0j), like="torch")
+        expected = expected_TRX_circ_expval_jacobians(phi.detach().numpy(), subspace)
+        assert math.allclose(jacobian.detach().numpy(), expected)
+
+    @pytest.mark.tf
+    @pytest.mark.parametrize("subspace", [(0, 1), (0, 2)])
+    def test_tf_results_and_backprop(self, subspace):
+        """Tests execution and gradients of a basic circuit using tensorflow."""
+        import tensorflow as tf
+
+        phi = tf.Variable(4.873)
+
+        dev = DefaultQutritMixed()
+
+        with tf.GradientTape(persistent=True) as grad_tape:
+            qs = self.get_TRX_quantum_script(phi, subspace)
+            result = dev.execute(qs)
+
+        expected = expected_TRX_circ_expval_values(phi, subspace)
+        assert qml.math.allclose(result, expected)
+
+        expected = expected_TRX_circ_expval_jacobians(phi, subspace)
+        assert math.all(
+            [
+                math.allclose(grad_tape.jacobian(one_obs_result, [phi])[0], one_obs_expected)
+                for one_obs_result, one_obs_expected in zip(result, expected)
+            ]
+        )
+
+    @pytest.mark.tf
+    @pytest.mark.parametrize("op,param", [(qml.TRX, np.pi), (qml.QutritBasisState, [1])])
+    def test_qnode_returns_correct_interface(self, op, param):
+        """Test that even if no interface parameters are given, result's type is the correct
+        interface."""
+        dev = DefaultQutritMixed()
+
+        @qml.qnode(dev, interface="tf")
+        def circuit(p):
+            op(p, wires=[0])
+            return qml.expval(qml.GellMann(0, 3))
+
+        res = circuit(param)
+        assert qml.math.get_interface(res) == "tensorflow"
+        assert qml.math.allclose(res, -1)
+
+    def test_basis_state_wire_order(self):
+        """Test that the wire order is correct with a basis state if the tape wires have a
+        non-standard order."""
+        dev = DefaultQutritMixed()
+
+        tape = qml.tape.QuantumScript(
+            [qml.QutritBasisState([2], wires=1), qml.TClock(0)], [qml.state()]
+        )
+        expected_vec = np.array([0, 0, 1, 0, 0, 0, 0, 0, 0], dtype=np.complex128)
+        expected = np.outer(expected_vec, expected_vec)
+        res = dev.execute(tape)
+        assert qml.math.allclose(res, expected)
+
+
+@pytest.mark.parametrize("subspace", [(0, 1), (0, 2)])
+class TestSampleMeasurements:
+    """Tests circuits using sample-based measurements.
```

Comment:
Like with `test_default_qubit.py` I reuse tests from sampling, simulate, and measurement to make sure functionality doesn't change through device.

Reply:
This is a lot of duplication, if nothing else, does it make sense to just inherit from that class so that you don't have to duplicate all the static methods? (PL team: is that fine convention-wise? other suggestions here?)

Reply:
We don't generally inherit from test classes, so I don't personally mind having this duplication in tests. However, in the future, we can consider collecting the tests into one file and parametrizing over the device. This type of duplication is rampant in the `tests/interfaces` module, where the test files for each interface have the same tests, with the only significant difference being the function used to compute gradients/jacobians, as those are dependent on the interface being used.

Reply:
Alright I will leave it to for the future